src/classes/RD2_VisualizeScheduleController.cls

```diff
@@ -151,7 +156,17 @@ public with sharing class RD2_VisualizeScheduleController {
                 'StartDate__c, ' +
                 'EndDate__c ' +
                 'FROM RecurringDonationSchedules__r)';
+
+            String oppSubquery =
+                '(SELECT CloseDate, ' +
+                'IsClosed, IsWon ' +
+                'FROM npe03__Donations__r ' +
+                'WHERE IsClosed = false ' +
+                'OR CloseDate >= TODAY ' +
```

Comment:
We need to get the matching Opp within the Next Donation Date range to see if it is closed and use that to check how many outstanding installments should exist.